src/D2L.CodeStyle.Analyzers/Extensions/Microsoft.CodeAnalysis.cs

```diff
@@ -150,6 +150,31 @@ internal static partial class RoslynExtensions {
 			return false;
 		}
 
+		public static bool IsTypeMarkedWith( 

```

Comment:
We have so many of these sort of extension methods!!! :'(

Reply:
To be fair, I wrote this one so that we would stop writing IsTypeMarkedImmutable, IsTypeMarkedSingleton, etc.  You would just say IsTypeMarkedWith( Immutable ), IsTypeMarkedWith( Singleton ), but I also thought that modifying the existing code to use that would be outside the scope of this PR.